src/components/input/Input.vue

```diff
@@ -218,6 +240,47 @@ export default {
             this.$nextTick(() => {
                 this.$refs.input.focus()
             })
+        },
+		
+		shouldUpdateLocalValue (localValue, checkVal, fromBlur = false) {
+            return this.isNotInFocusAndDirty(localValue, checkVal, fromBlur)||this.isDirtyWithModifiers(localValue, checkVal, fromBlur);
+        },
+		
+        isNotInFocusAndDirty (localValue, checkVal, fromBlur) {
+            if(!fromBlur){//from value (v-model) watcher
+                if(this.isFocused && !this.forcedSync){
+                    if(this.trim||this.number)return false;//don't update value if it's trim/number and it's still focused
+                }
+                return true;//update value if it's trim/number and unfocused
+            }
+            //sync local variable on blur, default v-model behaviour
+            return localValue !== checkVal;
+        },
+		
+        isDirtyWithModifiers (localValue, checkVal, fromBlur) {//if input have modifiers
+            if (this.number) {
+                return this.toNumber(localValue) !== this.toNumber(checkVal);//update value if it's num
+            }
+            else if(this.trim){
+                return localValue.trim() !== localValue.trim();//update value if it's trim
+            }
+            //sync local variable on blur, default v-model behaviour, seems like checks before isnot needed
+            return localValue !== checkVal;
+        },
+		
+        toNumber (val) {
+            let n = parseFloat(val);
+            return isNaN(n) ? val : n
+        },
+		
+		localOnBlur() {
+			if(this.syncOnBlur){
+				//force update local variable to value prop, same as $forceUpdate in v-model
+                //not sure about this check, but v-model doing this too
+				// if(this.shouldUpdateLocalValue(this.newValue,this.value, true)){
```

Comment:
v-model adding this checks too, but it's can be heavy, that's why this is commented. So, if this.syncOnBlur then we should instantly sync value on blur.